### DISCUSSION TITLE
Is BFV scheme going to be removed?

### DISCUSSION THREAD

**Discussion by fedejinich** [tuneinsight/lattigo#384] [General]
BFV it's being "unified" in this [PR](https://github.com/tuneinsight/lattigo/pull/309). So, will the BFV scheme no longer be supported in lattigo?

**Comment by Pro7ech** [comment]
The BFV scheme will not be removed in the next release, but the back-end implementation will be entirely replaced by a wrapper over the package `lattigo/bgv`, which will implement a generalized variant of the BFV and BGV schemes. 

The functionalities, API and noise growth will stay identical to the current implementation (up to minor bug fixes and improvements).

You can check what is going to happen here: https://github.com/tuneinsight/lattigo/tree/dev_release_v4.2.0/bfv

**Reply by fedejinich** [reply]
Okay thank you for the answer!
I'm asking this because I've changed my current lattigo version (latest release) to that experimental branch, and performance improvement was incredible (from 20 seconds to 10 seconds). I'm benchmarking pasta-BFV transcipher for a 128 legnth byte array. I know that I'm not giving that much information but is this kind of situation expected? Does it have anything to do with the rns acceleration?

**Reply by Pro7ech** [reply]
Interesting, so you are telling me that there is on your side a 2x speed increase between the current v4.1.0 and the future v4.2.0?

If so, I am really curious. Maybe you could do a pprof between the two versions that you used, and tell me which low level operations got cut in half.

Did you also update your go version?
